app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt

```diff
@@ -86,28 +104,29 @@ class PhotoInfoActivity : AppCompatActivity() {
 
     override fun onSaveInstanceState(outState: Bundle?, outPersistentState: PersistableBundle?) {
         super.onSaveInstanceState(outState, outPersistentState)
-        outState?.putParcelableArray(LIST_OF_CLASSES_KEY, classButtons)
+        outState?.putParcelableArray(LIST_OF_CLASSES_KEY, classButtons.toTypedArray())
     }
 
     @OnClick(R.id.save_button)
     fun onSaveButtonPressed(view: View) {
-
         picNoteToEdit.title = title_input.text.toString()
         picNoteToEdit.description = description_input.text.toString()
-        // todo vpineda we need to create a dropdown for all of the classes or projects!
+        // todo vpineda remove all of the first clause, create menu to add new folders
         if (folder_input.text.isNotEmpty()) {
             picNoteToEdit.folderId = Integer.parseInt(folder_input.text.toString())
+        } else if (classButtons.any { it.active }) {
+            picNoteToEdit.folderId = classButtons.first { it.active }.id
+        }
 
-            // create new Folder
-            LaunchNoteDatabase.getDatabase(this)?.let {
-                it.folderDao().findById(folder_input.text.toString()).firstElement()
-                        .subscribeOn(Schedulers.io())
-                        .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe { loadedFolder ->
-                            // add picNoteToEdit to the folder with matching folderId
-                            addPicNoteToFolder(loadedFolder)
-                        }
-            }
+        // update folder or create folder for picnote
+        LaunchNoteDatabase.getDatabase(this)?.let {
```

Comment:
there's also the possibility that the input is empty, and no folder button's selected, so the PicNote just doesn't belong to a folder.